hiero-observability/hiero-metrics-core/src/main/java/org/hiero/metrics/api/core/MetricKey.java

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: Apache-2.0
+package org.hiero.metrics.api.core;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Objects;
+import org.hiero.metrics.api.utils.MetricUtils;
+
+/**
+ * A unique key for identifying a {@link Metric} by its name and type.
+ * Key instance is immutable and can be used to retrieve a metric from a {@link MetricRegistry}.
+ * <p>
+ * Callers may additionally categorize the metric by calling {@link #withCategory(String)}.
+ */
+public final class MetricKey<M extends Metric> {
```

Comment:
What's the purpose of this class? Are metric names unique or pairs (metric name, type) unique? In other words, can there be two metrics with the same name, but different types? If not, then each metric can be identified by a single String, no need to have a dedicated "key" class for that?